src/filters/authfilter.bal

```diff
@@ -0,0 +1,111 @@
+import ballerina/config;
+import ballerina/http;
+import ballerina/log;
+import ballerina/stringutils;
+
+
+
+public type AuthFilter object {
+
+    private http:Client clientEndpoint;
+
+    public function __init() {
+        self.clientEndpoint = new (config:getAsString("mosip.auth.base_url"));
+    }
+    *http:RequestFilter;
+
+    public function filterRequest(http:Caller caller, http:Request request,
+        http:FilterContext context) returns boolean {
+        if (config:getAsString("mosip.auth.filter_status", "disable") == "enable") {
+            http:Response errorsResponse = new;
+            if (request.getQueryParamValue("hub.mode") == "publish") {
+                http:Cookie[] cookies = request.getCookies();
+                http:Cookie[] authCookie = cookies.filter(function
+                    (http:Cookie cookie) returns boolean {
+                    return cookie.name == "Authorization";
+                });
+                if (authCookie.length() > 0) {
+                    string? authToken = authCookie[0].value;
+                    if (authToken is string) {
+                        http:Request req = new;
+
+                        req.addHeader("Cookie", "Authorization=".concat(authToken));
+                        log:printInfo("authtoken" + authToken);
```

Comment:
Can we make this debug log, since it is a sensitive info?

Reply:
sure